client.go

```diff
@@ -48,17 +51,45 @@ type Client struct {
 }
 
 func New(writeKey string) *Client {
+	return NewWithConfig(writeKey, Config{})
+}
+
+func NewWithConfig(writeKey string, config Config) *Client {
```

Comment:
What do you think if `Config` contains the writeKey?

```
func New(writeKey string) *Client {
  return NewWithConfig(Config{WriteKey: writeKey})
}

func NewWithConfig(config Config) *Client {
...
}
```

Reply:
Configs tend to imply optionality. The writekey is not optional. I am not sure it belongs in the config